### PR TITLE
Restric usage

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Depends on the fix in https://github.com/OP-TEE/optee_os/pull/6505

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
The failure is expected since there's an added test for something that is fixed in OP-TEE/optee_os#6505.

**Comment by jforissier** [comment]
For "ta: storage: fix ta_storage_cmd_key_in_persistent() cleanup":

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jforissier** [comment]
For "xtest: regression: add case 6021 object usage":

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Tags applied.

File: host/xtest/regression_6000.c [hunk 1]
```diff
@@ -2254,3 +2254,62 @@ static void xtest_tee_test_6020_single(ADBG_Case_t *c, uint32_t storage_id)
...
+	ADBG_EXPECT_COMPARE_UNSIGNED(c, obj_info.objectUsage, ==, 0xffffffff);
+
+	res = fs_restrict_usage(&sess, obj, 0xfffffffe);
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, res))
+		goto exit;
+
+	res = fs_get_obj_info(&sess, obj, &obj_info);
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, res))
+		goto exit;
+	ADBG_EXPECT_COMPARE_UNSIGNED(c, obj_info.objectUsage, ==, 0xfffffffe);
```
Comment (etienne-lms) [inline host/xtest/regression_6000.c:2290]: I would suggest to clear an explicitly meaningful usage bitflag:
```c
	ADBG_EXPECT_COMPARE_UNSIGNED(c, obj_info.objectUsage, ==,
				     0xffffffff & ~TEE_USAGE_EXTRACTABLE);

File: host/xtest/regression_6000.c [hunk 2]
```diff
@@ -2254,3 +2254,62 @@ static void xtest_tee_test_6020_single(ADBG_Case_t *c, uint32_t storage_id)
...
+
+exit:
+	ADBG_EXPECT_TEEC_SUCCESS(c, fs_unlink(&sess, obj));
+	TEEC_CloseSession(&sess);
+	return;
+}
+DEFINE_TEST_MULTIPLE_STORAGE_IDS(xtest_tee_test_6021)
+ADBG_CASE_DEFINE(regression, 6021, xtest_tee_test_6021,
+		 "Modify and check object usage");
```
Comment (etienne-lms) [inline host/xtest/regression_6000.c:2315]: Suggestion: "Modify and check **persistent** object usage"

File: host/xtest/regression_6000.c [hunk 3]
```diff
@@ -2254,3 +2254,62 @@ static void xtest_tee_test_6020_single(ADBG_Case_t *c, uint32_t storage_id)
 DEFINE_TEST_MULTIPLE_STORAGE_IDS(xtest_tee_test_6020)
 ADBG_CASE_DEFINE(regression, 6020, xtest_tee_test_6020,
 		 "Object IDs in SHM (negative)");
+
+static void xtest_tee_test_6021_single(ADBG_Case_t *c, uint32_t storage_id)
+{
+	const uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
+			       TEE_DATA_FLAG_ACCESS_WRITE |
+			       TEE_DATA_FLAG_ACCESS_WRITE_META;
```
Comment (etienne-lms) [inline host/xtest/regression_6000.c:2262]: `TEE_DATA_FLAG_OVERWRITE` to ensure object is destroyed and recreated?

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commit
"TEE_DATA_FLAG_OVERWRITE" with imperative mode in commit message.

File: host/xtest/regression_6000.c (hunk 3 above)
Comment (jenswi-linaro) [reply host/xtest/regression_6000.c:2262]: OK, I'll add it to the create call.

**Comment by jenswi-linaro** [comment]
Addressing comments, tag applied.

File: host/xtest/regression_6000.c [hunk 4]
```diff
@@ -2254,3 +2254,65 @@ static void xtest_tee_test_6020_single(ADBG_Case_t *c, uint32_t storage_id)
...
+
+	ou &= ~TEE_USAGE_EXTRACTABLE;
+	res = fs_restrict_usage(&sess, obj, ou);
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, res))
+		goto exit;
+
+	res = fs_get_obj_info(&sess, obj, &obj_info);
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, res))
+		goto exit;
+	ADBG_EXPECT_COMPARE_UNSIGNED(c, obj_info.objectUsage, ==, ou);
```
Comment (etienne-lms) [inline host/xtest/regression_6000.c:2293]: `goto exit` here on failure I think.

Comment (jenswi-linaro) [reply host/xtest/regression_6000.c:2293]: The test has failed at this point, but it's not completely useless to open a new session, reopen the object, and check once more. The test is stable even on errors.

Comment (etienne-lms) [reply host/xtest/regression_6000.c:2293]: Ok

**Comment by etienne-lms** [comment]
Sorry for the bad reference for my R-b tag.
Whatever,  `Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>` for both commits.

**Comment by jenswi-linaro** [comment]
Tag applied

**Pull request merged**